### ISSUE TITLE
[RFC] Identify, Discuss, and Generate a List of Useful Guides

### ISSUE THREAD

**Issue by possumbilities** [creativecommons/search#149]
## Problem

> These Guides are meant to serve as an expansion to existing documentation for beginners of this project, that means Guides should be directly and uniquely related to this project. Generalized beginner level material about broader topics like Github, Git, Command-line basics, etc. may be better served in the Open Source documentation as a part of our new overall general beginner guides.

## Description
* Please read the guides [README.md](https://github.com/creativecommons/search/tree/main/guides) which outlines what a new Guide should include, and the conventions it should follow. 
* Please comment below with any suggestions for Guides we might benefit from creating that are not already there. 
* This Issue can be referenced in sub-issues that spin out from it.

## Additional context
* #110
* #112
* #141

## Implementation

- [ ] I would be interested in implementing this feature.

**Comment by ImaJin14** [comment]
> I'm currently on it now!
> 

It's a discussion... Meaning you drop down your ideas here....

**Comment by yokwejuste** [comment]
> > I'm currently on it now!
> 
> It's a discussion... Meaning you drop down your ideas here....

Thanks for that @ImaJin14 !

I said I'm on it, meaning going to look for ideas for this now

**Comment by onyijne** [comment]
Talking about useful guides, I decided to contribute on CC Meta Search today, reading the internship task on Outreachy website it states interns are expected to 

> Rewrite the application, maintaining its original function, as a static site on GitHub pages, using semantic, standards-compliant, and accessible HTML, CSS, and JavaScript. Update the visual aesthetic and overall appearance to better correspond with other current CC web entities.

To achieve this I assume it means to rewrite the application to a JavaScript application, that can be hosted on Github pages while maintaining its current functions. This leads to the question if the [JavaScript guide](https://opensource.creativecommons.org/contributing-code/javascript-guidelines/) in the other [RFC](https://github.com/creativecommons/creativecommons.github.io-source/issues/674) should be adhered to when refactoring the codebase? If yes, then can it be refactored using Vue as the preferred JavaScript framework?

**Comment by possumbilities** [comment]
@onyijne I think the Javascript guide on the open source site needs a little clarification. The meaning there is more that if one is going to use a front-end component framework, with JS the standards at the moment, they should go with Vue (rather than React, Angular, Preact, etc.)

But that doesn't mean a project has to go that far in abstraction. A project can have Javascript that's functional and behavioral, and leaves the component rendering to mostly HTML and CSS without the need for introducing a f/e JS library.

So, _If_ the need calls for a f/e JS library, Vue is the route, but all projects utilizing JS don't necessarily need a f/e JS library to operate effectively.

**Comment by ImaJin14** [comment]
I think putting up a guide which contains proper links for setting up docker on various os  and links that help with various github task such as opening up a PR... Cloning a repository and syncing it with the remote branch



Those where the most noticed beginner issues I noticed and had to aid some people with

**Comment by PreciousOritsedere** [comment]
Adding a detailed guide on setting up Docker will be very helpful. We could also add the possible errors that could spring up and include links that could help resolve those errors

**Comment by JennySimen** [comment]
A step by step guide for setting up docker most especially for windows with images will be very helpful as it will be easier to follow and better to understand.

**Comment by dikehprosper** [comment]
A link to docker installation is already present in the readme.md…what we can really work on is stating out the possible errors, beginners might encounter while trying to set up the projects and proffer solutions to them. I would suggest this to be in a different folder

**Comment by JennySimen** [comment]
I think setting up a guide which explains how one can go about searching for issues to solve in the codebase and on the site will be of great help to new contributors and beginners.
Going through the contribution file, it explains how one can find issues from the issues list on the repo and how to create new ones but doesn't explain how you can search for them before creating. But in a case where all issues are taken or no issues available, it becomes abit confusing when you can't find one.
Having this guide will be of great help.

**Comment by mbonamensa** [comment]
Suggestions on this discussion are awesome and would be beneficial to newcomers. 
I think a guide that has a format for asking questions related to this project can be written as well or can be included in any of the above-mentioned guides as a postscript.

**Comment by possumbilities** [comment]
> I think setting up a guide which explains how one can go about searching for issues to solve in the codebase and on the site will be of great help to new contributors and beginners. Going through the contribution file, it explains how one can find issues from the issues list on the repo and how to create new ones but doesn't explain how you can search for them before creating. But in a case where all issues are taken or no issues available, it becomes abit confusing when you can't find one. Having this guide will be of great help.

This is more a general guide that already exists in some part on the [Contribution Guidelines](https://opensource.creativecommons.org/contributing-code). If anything there might be a need for that site to be modified or updated to address some specific concerns. But Guides here should be specific to this project, and while the above suggestion is useful, it's more general in application.

**Comment by possumbilities** [comment]
I agree that a Docker specific installation guide for Windows is likely useful, specifically for this project.

**Comment by mbonamensa** [comment]
A Frequently Asked Question (FAQ) can also be considered a guide. I am not sure about Slack but I have seen questions being asked on a Discord community being turned into a FAQ channel where people can go to search for questions or ask questions related to the project if they do not find the answers they are looking for. Like a way of documenting relevant questions so that there isn't a repetition of messages. Almost like a mini stack overflow, but for CC search 😁

**Comment by LOVKUSH9888** [comment]
Choose a project:You can find open source projects on websites such as GitHub, GitLab, orBitbucket. 

Familiarize yourself with the project:You can read the documentation and code base directly on the project'swebsite. 

Find an issue to work on:You can find open issues by visiting the project's issue tracker. 

Create a fork:Navigate to the project's repository and click the "Fork" button. 

Make changes:Clone the forked repository to your local machine:

git clone https://github.com/<your-username>/<project>.git
Change to the cloned directory:

bash 
cd <project>
Create a new branch to make your changes:




css 
git checkout -b <branch-name>
Make changes to the code.
Stage and commit your changes:

sql 
git add . git commit -m "<commit-message>"
Push the changes to your fork:

markdown 
git push origin <branch-name> 

Submit a Pull Request:Go to the project's repository on your forked repository on the website andclick the "Compare & pull request" button.
Fill out the pull request form and submit it. 


Collaborate with the community:You can participate in discussions and help others through the project's issuetracker or community forums.

7. 
Note: The exact commands may vary depending on the version control system used bythe project (e.g. Git, Mercurial, etc.), the website hosting the project (e.g. GitHub, GitLab,Bitbucket, etc.), and the operating system you are using.6.
